src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -118,6 +124,76 @@ public synchronized void close() throws IOException {
             return;
         }
 
+        try {
+            stopGracefully();
+        } finally {
+            stopForcefully();
+        }
+    }
+
+    @SuppressFBWarnings("SWL_SLEEP_WITH_LOCK_HELD")
+    private void stopGracefully() {
+        //TODO: Temporarily limiting graceful shutdown to only processes that are created using ProcessImplForWin32
+        if (!(process instanceof ProcessImplForWin32)) {
+            return;
+        }
+        int pid = process instanceof ProcessImplForWin32
+                ? ((ProcessImplForWin32) process).getPid() : Processes.newPidProcess(process).getPid();
+        boolean sentConsoleCtrlEvent = false;
+        synchronized (Kernel32.INSTANCE) {
+            Kernel32 k32 = Kernel32.INSTANCE;
+            // if the console is already attached, a second attach call will fail.
+            if (!k32.FreeConsole()) {
+                logger.error("FreeConsole error {}", k32.GetLastError());
+            }
```

Comment:
if the console is not attached, does this log an unnecesary error?

Reply:
FreeConsole only returns false when there is an error, calling  FreeConsole when the console is already detached will return true.